third_party/docfx-doclet-143274/src/main/java/com/microsoft/build/Lookup.java

```diff
@@ -12,14 +12,17 @@
 
 public class Lookup {
 
-    private Map<String, String> globalLookup = new HashMap<>();
-    private Map<String, Map<String, String>> localLookupByFileName = new HashMap<>();
+    private static final int INITIAL_CAPACITY = 10000;
+    private final Map<String, String> globalLookup;
+    private final Map<String, Map<String, String>> localLookupByFileName;
 
     private final String UID_PACKAGE_NAME_REGEXP = "^.*?\\.(?=[A-Z].*)";
     private final String PARAM_PACKAGE_NAME_REGEXP = "(?<=[\\( ]).*?(?=[A-Z].*)";
     private final String METHOD_PARAMS_REGEXP = "\\s[^\\s]+?(?=[,)])";
 
     public Lookup(List<MetadataFile> packageMetadataFiles, List<MetadataFile> classMetadataFiles) {
+        this.globalLookup = new HashMap<>(INITIAL_CAPACITY);
+        this.localLookupByFileName = new HashMap<>(INITIAL_CAPACITY);

```

Comment:
Set a large initial capacity to *try* and reduce the number of copying done (capacity expansion)